src/Balsam/src/Balsam.Api/HubClient.cs

```diff
@@ -89,67 +96,106 @@ private async Task<bool> ProjectExists(string projectName)
             return true;
         }
 
-        public async Task<BalsamProject?> CreateProject(string preferredName, string description, string defaultBranchName)
+        public async Task<BalsamProject?> CreateProject(string preferredName, string description, string defaultBranchName, string username)
         {
             //Check if there is a program with the same name.
             if (await ProjectExists(preferredName))
```

Comment:
Skulle vi kunna slimma ner ProjectExists så att den inte behöver läsa in branches och dess props?